src/components/OcButton.spec.js

```diff
@@ -0,0 +1,109 @@
+import OcButton from './OcButton.vue';
+import {createLocalVue, shallowMount} from '@vue/test-utils';
+const localVue = createLocalVue()
+
+describe('OcButton', () => {
+  const testSlots = { default: '<p class="text">Test button</p>' }
+  const route = {
+    path: '/some/path',
+    name: 'some-route'
+  }
+  function getWrapperWithProps(props) {
+    return shallowMount(OcButton, { localVue: localVue, propsData: props })
+  }
+  function getWrapperWithTestSlot() {
+    return shallowMount(OcButton, { localVue: localVue, slots: testSlots })
+  }
+
+  it('displays slot html', () => {
+    const wrapper = getWrapperWithTestSlot()
+    const slot = wrapper.find('p')
+    expect(slot).toBeTruthy()
+    expect(slot.attributes('class')).toBe('text')
+    expect(slot.text()).toBe('Test button')
+  })
+
+  it('can stop class propagation', () => {
+    const wrapper = getWrapperWithProps({
+      stopClassPropagation: true,
+      size: 'middle'
+    })
+    expect(wrapper.attributes('class')).toContain('')
+  })
+
+  describe('emit click event', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it('emits click event when click is triggered', async () => {
+      await wrapper.trigger('click')
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeTruthy()
+    })
+    it('does not emit click event when input is triggered', async () => {
+      await wrapper.trigger('input')
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+    })
+  })
+  describe('disabled', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({ disabled: true })
+    })
+    it('should have disabled attribute set to disabled', () => {
+      expect(wrapper.attributes('disabled')).toBe('disabled')
+    })
+    it('does not emit click event', async () => {
+      await wrapper.trigger('click')
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+    })
+  })
+  describe('different types of button', () => {
+    it.each`
+      type             | expectLink | expectButton | expectRouterLink
+      ${'a'}           | ${true}    | ${false}     | ${false}
+      ${'button'}      | ${false}   | ${true}      | ${false}
+      ${'router-link'} | ${false}   | ${false}     | ${true}
+    `
+    ('can behave as a $type', (
+      {type, expectLink, expectButton, expectRouterLink}) => {
+        const wrapper = getWrapperWithProps({ type: type })
+        expect(wrapper.find('a').exists()).toBe(expectLink)
+        expect(wrapper.find('button').exists()).toBe(expectButton)
+        expect(wrapper.find('router-link-stub').exists()).toBe(expectRouterLink)
```

Comment:
click events should only be emitted for buttons https://github.com/owncloud/owncloud-design-system/blob/master/src/components/OcButton.vue#L180-L182
you could add a test for that here

Reply:
tests added